lib/Alcaeus/MongoDbAdapter/AbstractCursor.php

```diff
@@ -276,7 +276,7 @@ protected function getOptions($optionNames = null)
             $converter = 'convert' . ucfirst($option);
             $value = method_exists($this, $converter) ? $this->$converter() : $this->$option;
 
-            if ($value === null) {
+            if ($value === null || ($option === 'projection' && $value === [])) {
```

Comment:
Line 277 checks to see if a `convert<Option>` method exists. This special case should be handled in a `convertProjection` method.